sql/sql_select.cc

```diff
@@ -720,16 +720,19 @@ void vers_select_conds_t::print(String *str, enum_query_type query_type) const
 int period_update_condition(THD *thd, TABLE_LIST *table, SELECT_LEX *select,
                             vers_select_conds_t &conds, bool timestamp)
 {
+  DBUG_ASSERT(table);
+  DBUG_ASSERT(table->table);
 #define newx new (thd->mem_root)
-  TABLE_SHARE::period_info_t *period = table->table->s->get_period(conds.name);
+  TABLE_SHARE *s= table->table->s;
+  TABLE_SHARE::period_info_t *period= s->get_period(conds.name);
 
   if (!period)
   {
     my_error(ER_PERIOD_NOT_FOUND, MYF(0), conds.name.str);
     return -1;
   }
-  const LEX_CSTRING &fstart= period->start_field()->field_name;
-  const LEX_CSTRING &fend= period->end_field()->field_name;
+  const LEX_CSTRING &fstart= period->start_field(s)->field_name;
+  const LEX_CSTRING &fend= period->end_field(s)->field_name;
```

Comment:
The obnoxious thing here is that we're now supposed to write smth like `table->s->period->start_field(table->s)` to get a period start field of table share